### PR TITLE
Deprecate `names` for composite types in favor of `fieldnames`

### LINKED ISSUE #10325: Rename `names` to `fields` or `fieldnames`?
It has been brought up in a couple of discussions that it would be useful to overload the meaning of `names` for user defined types, but it's current meaning in `Base` (return a array of symbols for field names of a given type) kind of prevents this (you would assume `getfield(T, fieldnames(T)[1])` works).

Possible suggestions were to rename `names` to `fields` or `fieldnames`.

### PR DESCRIPTION
**Pull request by jakebolewski**
Closes #10325.

### CODE REVIEW

**Comment by JeffBezanson** [comment]
I thought the idea was to use fieldnames just to mean the field names of a composite type, and `names` for other things, like the exported names of a module.

**Comment by jakebolewski** [comment]
Yeah, I was not really sure about that.  I'll change it.

**Comment by jakebolewski** [comment]
The module changes have been removed.

**Comment by ivarne** [comment]
Looks good, but this should be in NEWS.md

File: base/sparse/cholmod.jl [hunk 1]
```diff
@@ -797,7 +797,7 @@ sparse(D::Dense)    = sparse(Sparse(D))
 # Calculate the offset into the stype field of the cholmod_sparse_struct and
 # change the value
 function change_stype!(A::Sparse, i::Integer)
-    offset = fieldoffsets(C_Sparse)[names(C_Sparse) .== :stype][1]
+    offset = fieldoffsets(C_Sparse)[9] # field offset of C_Sparse.stype
```
Comment (tkelman) [inline base/sparse/cholmod.jl]: no one is going to remember to change this...

File: NEWS.md [hunk 2]
```diff
@@ -28,7 +28,7 @@ New language features
 
 Language changes
 ----------------
-
+
```
Comment (ivarne) [inline NEWS.md]: What? 

Adding trailing whitespace?

Comment (jakebolewski) [inline NEWS.md]: shamed :(

**Comment by JeffBezanson** [comment]
For shame, there is no doc update here.

**Pull request merged**